src/AppInstallerCLICore/Workflows/ImportExportFlow.cpp

```diff
@@ -0,0 +1,204 @@
+            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_NO_APPLICATIONS_FOUND);
+        }
+    }
+
+    void SearchPackagesForImport(Execution::Context& context)
+    {
+        auto availableSources = Repository::GetSources();
+        std::vector<std::shared_ptr<IPackageVersion>> packagesToInstall;
+
+        // Aggregated source with all the required sources.
+        // We keep it as the source of the root context to keep all the
+        // source objects alive for install.
+        auto aggregatedSource = std::make_shared<Repository::CompositeSource>("*ImportSource");
+
+        // Look for the packages needed from each source independently.
+        // If a package is available from multiple sources, this ensures we will get it from the right one.
+        for (auto& requiredSource : context.Get<Execution::Data::PackageCollection>().RequestsFromSources)
+        {
+            if (!requiredSource.Details.Name.empty())
+            {
+                // For packages that come from a specific source, find the matching available source.
+                // Match required sources with the available sources by their arguments, as they may have been added with a different name.
+                // TODO: Can we do this with identifiers without opening the source?
+                std::optional<SourceDetails> matchingSource = {};
+                for (auto& availableSource : availableSources)
+                {
+                    if (availableSource.Arg == requiredSource.Details.Arg)
+                    {
+                        matchingSource = availableSource;
+                        break;
+                    }
+                }
+
+                if (matchingSource.has_value())
+                {
+                    requiredSource.Details.Name = matchingSource.value().Name;
+                }
+                else
+                {
+                    // TODO: Add option for ignoring/installing missing sources?
+                    AICLI_LOG(CLI, Error, << "Missing required source: " << requiredSource.Details.Name);
+                    AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_SOURCE_NAME_DOES_NOT_EXIST);
+                }
+            }
+
+            context << OpenNamedSource(requiredSource.Details.Name);
+            auto source = context.Get<Execution::Data::Source>();
+            aggregatedSource->AddAvailableSource(source);
+            for (const auto& packageRequest : requiredSource.Packages)
+            {
+                // Search for all the packages in the source
+                Logging::SubExecutionTelemetryScope subExecution;
+
+                // We want to do best effort to install all packages regardless of previous failures
+                auto searchContextPtr = context.Clone();
+                Execution::Context& searchContext = *searchContextPtr;
+                searchContext.Add<Execution::Data::Source>(source);
+
+                // TODO: Case insensitive?
```

Comment:
Finding the appropriate package and version to install is going to be far more complex than this.  We may not have the Id handy; we may just have the name and publisher and will require the ability to find something based on that.

But I think that having a review on the spec and schema will help solidify the requirements.